### PR TITLE
Update Location Cookie System

### PR DESCRIPTION
**Pull request by nssafir**
Formerly PR #94 but changed PR because there were many differences in a previous push. Now only have cookies for location object (derived from user coordinates) and address. Also speeds up time by making the user's location object on the homepage before the user hits 'submit'.

### CODE REVIEW

File: src/main/java/com/google/sps/data/UrlRequest.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
    */
   public static String urlQuery(String basePath, Map<String, String> params) throws IOException {
     String path = String.format("%s?%s", basePath, getParamsString(params));
+    System.out.println(path);
```
Comment (juanuribe28) [inline src/main/java/com/google/sps/data/UrlRequest.java]: [Nit] Remove test

File: src/main/java/com/google/sps/servlets/NewsServlet.java [hunk 2]
```diff
@@ -37,13 +37,16 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     List<String> terms = Arrays.asList(request.getParameterValues("key"));
+    String country = request.getParameter("country");
     HashMap<String, List<Article>> articleMap = new HashMap<>();
     terms.forEach((term) -> {
       try {
         Map<String, String> paramMap = new HashMap<>();
         paramMap.put("q", UrlRequest.encodeTerm(term));
         // TODO: Add lat and lng params to this servlet request before un-commenting line below.
```
Comment (juanuribe28) [inline src/main/java/com/google/sps/servlets/NewsServlet.java]: Remove TODO if completed.

File: src/main/webapp/index.html [hunk 3]
```diff
@@ -68,5 +68,6 @@
     <script src="scripts/cookies.js"></script>
     <script src="scripts/index.js"></script>
     <script src="/scripts/keywords.js"></script>
+    <script src="/scripts/toggle.js"></script>
```
Comment (juanuribe28) [inline src/main/webapp/index.html]: Why are you adding toggle.js to this file?

File: src/main/webapp/scripts/cookies.js [hunk 4]
```diff
@@ -82,3 +82,20 @@ function loadCheckboxStateCookie(cookieName) {
...
+      const lat = position.coords.latitude;
+      const lng = position.coords.longitude;
+      locationObj = await loadObject(`${LOCATION_OBJ_URL}?lat=${lat}&lng=${lng}`);
+      setCookie('location', JSON.stringify(locationObj), 1);
+    }, (err) => {
+      console.warn(`ERROR(${err.code}): ${err.message}`);
+      alert("We cannot access your location. Try checking your browswer settings");
+    });
+  }
+}
```
Comment (juanuribe28) [inline src/main/webapp/scripts/cookies.js]: [Nit] Add new line.

**Review by juanuribe28 — CHANGES_REQUESTED** [review]

**Review by juanuribe28 — APPROVED** [review]

**Review by daperlmut — APPROVED** [review]

**Pull request merged**